### PR TITLE
mac: Add SKIP_ZSH flag

### PR DESCRIPTION
**Pull request by chub**
When SKIP_ZSH is specified:
- do not install the 'zsh' homebrew package,
- do not set the ~/.zshrc file,
- do not attempt to switch to zsh at the end of the installation, and
- prompt the user to manually modify their shell's rc file.

At the moment, the flag is not overtly documented as the installation
instructions recommend all engineers to read the script before execution.

### CODE REVIEW

File: mac [hunk 1]
```diff
@@ -3,6 +3,16 @@
 # Be prepared to turn your laptop (or desktop, no haters here)
 # into an awesome development machine.
 
+SKIP_ZSH=0
+SHELL_CHANGES=()
+while getopts "z" opt; do
```
Comment (nathandcornell) [inline mac:8]: This seems a little counter-intuitive to me, option '-z' means do _not_ install zsh.
Maybe we could have a more verbose flag like '--no-zsh' instead?

File: mac [hunk 2]
```diff
@@ -210,5 +236,16 @@ if [ -f "$HOME/.laptop.local" ]; then
   . "$HOME/.laptop.local"
 fi
 
-SHELL=$( which zsh )
-$SHELL -l
+fancy_echo "Script complete!"
+
+if [[ "${SKIP_ZSH}" == 1 ]]; then
+  fancy_echo "Since zsh was not installed, please add the following to your shell rc:"
```
Comment (nathandcornell) [inline mac:242]: What do you think about just adding the changes to `~/.profile` instead? It should be loaded regardless of shell choice, and is POSIX-compliant:
https://superuser.com/a/789465

**Review by nathandcornell — CHANGES_REQUESTED** [review]
Looking good!
I just have a couple questions

File: mac (hunk 1 above)
Comment (chub) [reply mac:8]: I was feeling lazy about having to parse long options manually. =)

Could I use `-Z` to mean "skip zsh", and introduce `-z` as a no-op flag?

File: mac (hunk 2 above)
Comment (chub) [reply mac:242]: Hm... I could move it into `.profile`.  Is this script ever run in existing environments (with the changes already in `.zshrc`)? Is that something that needs to be supported?

We'll need to remember a few other things too:
- some people's stuff don't source `.profile` (I make a rule not to, so maybe the warning message isn't a bad idea); and
- additional scripts in the future may misbehave when it's a non-interactive shell (luckily `~/.asdf/asdf.sh` checks out).

File: mac (hunk 1 above)
Comment (nathandcornell) [reply mac:8]: Sounds reasonable to me!

File: mac (hunk 2 above)
Comment (nathandcornell) [reply mac:242]: I don't think that anyone ever re-runs this script, but that does add to the complexity.

I originally thought that `~.profile` would _always_ be evaluated, but if that's not true, we shouldn't rely on it.

Perhaps just echoing out the needed changes is good enough for now, we can always add more automation later.

File: mac (hunk 1 above)
Comment (chub) [reply mac:8]: Done!

**Comment by chub** [comment]
Whoops. My pre-merged history-modifying workflow doesn't make github PRs happy. Follow-up here: https://github.com/Flatbook/laptop/pull/8

**Pull request closed without merging**